src/corehost/cli/runtime_config.h

```diff
@@ -18,45 +19,110 @@ enum class roll_fwd_on_no_candidate_fx_option
     major_or_minor
 };
 
+class runtime_config_framework_t
+{
+public:
+    // Uses a "nullable<T>" pattern until we add such a type
+
+    runtime_config_framework_t()
+        : has_fx_ver(false)
+        , has_roll_fwd_on_no_candidate_fx(false)
+        , has_patch_roll_fwd(false)
+        , fx_ver(_X(""))
+        , patch_roll_fwd(false)
+        , roll_fwd_on_no_candidate_fx((roll_fwd_on_no_candidate_fx_option)0)
+        { }
+
+    const pal::string_t* get_fx_ver() const
+    {
+        return (has_fx_ver ? &fx_ver : nullptr);
+    }
+    void set_fx_ver(pal::string_t value)
+    {
+        has_fx_ver = true;
+        fx_ver = value;
+    }
+
+    const bool* get_patch_roll_fwd() const
+    {
+        return (has_patch_roll_fwd ? &patch_roll_fwd : nullptr);
+    }
+    void set_patch_roll_fwd(bool value)
+    {
+        has_patch_roll_fwd = true;
+        patch_roll_fwd = value;
+    }
+
+    const roll_fwd_on_no_candidate_fx_option* get_roll_fwd_on_no_candidate_fx() const
+    {
+        return (has_roll_fwd_on_no_candidate_fx ? &roll_fwd_on_no_candidate_fx : nullptr);
+    }
+    void set_roll_fwd_on_no_candidate_fx(roll_fwd_on_no_candidate_fx_option value)
+    {
+        has_roll_fwd_on_no_candidate_fx = true;
+        roll_fwd_on_no_candidate_fx = value;
+    }
+
+private:
+    bool has_fx_ver;
+    bool has_patch_roll_fwd;
+    bool has_roll_fwd_on_no_candidate_fx;
+
+    pal::string_t fx_ver;
+    bool patch_roll_fwd;
+    roll_fwd_on_no_candidate_fx_option roll_fwd_on_no_candidate_fx;
+};
+
+class runtime_config_t;
```

Comment:
Is this necessary?

Reply:
Yes it's a forward declaration since runtime_config_t in parse(...) uses its own class

Reply:
didn't parse(..) use its own class before this change?  How did that work?

The original code was:

```C++
class runtime_config_t
{
public:
    runtime_config_t();
    void parse(const pal::string_t& path, const pal::string_t& dev_path, const runtime_config_t* defaults);
```

Reply:
You're correct; that forward decl was not necessary. Removing.